packages/network-controller/src/NetworkController.ts

```diff
@@ -896,6 +1041,260 @@ export class NetworkController extends BaseControllerV2<
+   * networks as well as the custom networks in state.
+   *
+   * @returns The network clients keyed by ID.
+   */
+  #createAutoManagedNetworkClientRegistry() {
+    const infuraNetworkClientRegistry =
+      this.#buildIdentifiedInfuraNetworkClientConfigurations().reduce(
+        (obj, [networkClientId, networkClientConfiguration]) => {
+          const autoManagedNetworkClient = createAutoManagedNetworkClient(
+            networkClientConfiguration,
+          );
+          return { ...obj, [networkClientId]: autoManagedNetworkClient };
+        },
+        {} as AutoManagedInfuraNetworkClientRegistry,
+      );
+
+    const customNetworkClientRegistry =
+      this.#buildIdentifiedCustomNetworkClientConfigurations().reduce(
+        (obj, [networkClientId, networkClientConfiguration]) => {
+          const autoManagedNetworkClient = createAutoManagedNetworkClient(
+            networkClientConfiguration,
+          );
+          return { ...obj, [networkClientId]: autoManagedNetworkClient };
+        },
+        {} as AutoManagedCustomNetworkClientRegistry,
+      );
+
+    const [networkClientType, networkClientId, networkClientConfiguration] =
+      this.#buildIdentifiedNetworkClientConfigurationFromProviderConfig();
+
+    if (networkClientType === NetworkClientType.Infura) {
+      infuraNetworkClientRegistry[networkClientId] =
+        createAutoManagedNetworkClient(networkClientConfiguration);
+    } else {
+      customNetworkClientRegistry[networkClientId] =
+        createAutoManagedNetworkClient(networkClientConfiguration);
+    }
+
+    return {
+      [NetworkClientType.Infura]: infuraNetworkClientRegistry,
+      [NetworkClientType.Custom]: customNetworkClientRegistry,
+    };
+  }
+
+  /**
+   * Constructs the list of network clients for Infura networks (that is,
+   * those that we know Infura supports).
+   *
+   * @returns The network clients.
+   */
+  #buildIdentifiedInfuraNetworkClientConfigurations(): [
+    string,
+    InfuraNetworkClientConfiguration,
+  ][] {
+    return knownKeysOf(InfuraNetworkType).map((network) => {
+      const networkClientId = buildInfuraNetworkClientId(network);
+      const networkClientConfiguration: InfuraNetworkClientConfiguration = {
+        type: NetworkClientType.Infura,
+        network,
+        infuraProjectId: this.#infuraProjectId,
+      };
+      return [networkClientId, networkClientConfiguration];
+    });
+  }
+
+  /**
+   * Constructs the list of network clients for custom networks (that is, those
+   * that have been added to `networkConfigurations` in state).
+   *
+   * @returns The network clients.
+   */
+  #buildIdentifiedCustomNetworkClientConfigurations(): [
+    string,
+    CustomNetworkClientConfiguration,
+  ][] {
+    return Object.entries(this.state.networkConfigurations).map(
+      ([networkConfigurationId, networkConfiguration]) => {
+        if (networkConfiguration.chainId === undefined) {
```

Comment:
You had commented on this before when it was in `#createAutoManagedNetworkClientRegistry`; now it's here.

These checks previously lived in `setActiveNetwork` — they were added when that method was added along with some tests — so I moved them over to make the tests pass. I presume this is here to verify that the extension does the right thing, though I'm not sure. In general though do we need these kinds of checks? I feel like we hadn't been for a while, and then some slipped in earlier on in the multichain project.

Reply:
Ah I see, I hadn't realized they were ported. Yes I would assume they were added in the extension as a precaution due to the lack of types.